tests/integration/level_depth/test_search_different_depths.py

```diff
@@ -45,13 +45,13 @@ def validate_granularity_1(resp):
         'I am chunk 0 of doc 2,',
         ' I am chunk 3 of doc 3',
     ]
-
-    search_flow = Flow.load_config('flow-query.yml')
-    with search_flow:
+    response_mock = mocker.Mock(wrap=validate_granularity_1)
+    with Flow.load_config('flow-query.yml') as search_flow:
         search_flow.search(
             input_fn=search_data,
-            output_fn=validate_granularity_1,
+            output_fn=response_mock,
             callback_on='body',
         )
 
     del os.environ['JINA_TEST_LEVEL_DEPTH_WORKSPACE']
+    response_mock.assert_called()
```

Comment:
I have doubled checked and think we are doing it wrong. It should be:
```
response_mock = mocker.Mock()
...
    output_fn=response_mock

for arguments in response_mock.call_args_list:
    validate_granularity_1(*arguments)
```
With this we get rid of the assertion stack trace, since the assert happens outside of the flow.

Reply:
What if the `callback` is called more than once? When you provide more than one requests in input?